### PR TITLE
Use paranoid mode in tests

### LINKED ISSUE #1080: make PARANOID_MODE do something in tests
"I would like to mention here that this bug was not catched by our unit tests. ManageDataTests.cpp contains tests for adding, changing and modyfing DATA ledger entries, it even runs with PARANOID_MODE=true. But unfortunately, it never closes the ledger which causes checkAgainstDatabase to be called. So the PARANOID_MODE=true has no effects on these tests at all. I think that either applyCheck() should close a ledger with only one transaction (I've tried that, but there are some issues to be resolved), or at least LedgerDelta::checkAgainstDatabase() should be called manually around app.getLedgerManager().checkDbState(); call (tried that one as well, works good and finds the bug mentioned above). I'll be happy with either change, but closing ledger seems to test behavior matching live node more closely." - rafal

### PR DESCRIPTION
**Pull request by vogel**
Fixes #1080

### CODE REVIEW

File: src/ledger/LedgerDelta.cpp [hunk 1]
```diff
@@ -417,13 +417,8 @@ LedgerDelta::markMeters(Application& app) const
 }
 
 void
-LedgerDelta::checkAgainstDatabase(Application& app) const
+LedgerDelta::checkAgainstDatabase(Database& db) const
 {
-    if (!app.getConfig().PARANOID_MODE)
```
Comment (jedmccaleb) [inline src/ledger/LedgerDelta.cpp]: Why did you remove this check? If you want this mode in the tests just set it in the config.

Comment (vogel) [inline src/ledger/LedgerDelta.cpp]: This check is moved to: https://github.com/stellar/stellar-core/pull/1161/files#diff-26d755120733c7309bb85f71f8fcd2c9R207

The whole problem was that PARANOID_MODE was enabled in all tests config, but was ignored as implementation of applyCheck in tests did something a bit different than normal closeLedger. I've modified applyCheck to work exactly as closeLedger (by calling it) when possible (when Transaction is valid), so we won't encounter that kind of bug again.

Also applyCheck requires some additional checks of LedgerDelta (starts with verify "modified accounts invariants"), so I've added parameter to closeLedger to perform these checks. It was also a good place to extract checkAgainstDatabase() from that place and move check for PARANOID_MODE up in the ownership chain. This removed dependency on app.getConfig() from LedgerDelta and made checkAgainstDatabase() do what it says in the name (before that this methods should be named checkAgainstDatabaseIfParanoidMode()).
